lib/logstash/inputs/kinesis.rb

```diff
@@ -49,8 +51,15 @@ class LogStash::Inputs::Kinesis < LogStash::Inputs::Base
   # to enable the cloudwatch integration in the Kinesis Client Library.
   config :metrics, :validate => [nil, "cloudwatch"], :default => nil
 
-  def initialize(params = {}, kcl_class = KCL::Worker)
-    @kcl_class = kcl_class
+  #what a nasty hack to use the overloaded method :_(
+  java_import KCL::Worker::Builder
+  class Builder
+    #I don't know how to make it work using the class directly
+    java_alias :v2RecordProcessorFactory, :recordProcessorFactory, [Java::JavaClass.for_name(KCL_PROCESSOR_FACTORY_CLASS)]
```

Comment:
Ok! Thanks for your help!

I think even changing the tests, you will need to do the alias thing
because the builder needs that you set the processor factory.

Without the alias, in my machine the jvm warned me about the ambiguous
method and picked the v2 implementation. In the container I used for
testing the jvm warned the same message but picked the v1 method. Even
using the same ruby version (2.2.1).

Let me know if you need that I change anything.

Thank you! 😊
El El lun, 28 sept 2015 a las 21:40, Brian Palmer <notifications@github.com>
escribió:

> In lib/logstash/inputs/kinesis.rb
> <https://github.com/codekitchen/logstash-input-kinesis/pull/4#discussion_r40595584>
> :
>
> > @@ -49,8 +51,15 @@ class LogStash::Inputs::Kinesis < LogStash::Inputs::Base
> >    # to enable the cloudwatch integration in the Kinesis Client Library.
> >    config :metrics, :validate => [nil, "cloudwatch"], :default => nil
> >
> > -  def initialize(params = {}, kcl_class = KCL::Worker)
> > -    @kcl_class = kcl_class
> > +  #what a nasty hack to use the overloaded method :_(
> > +  java_import KCL::Worker::Builder
> > +  class Builder
> > +    #I don't know how to make it work using the class directly
> > +    java_alias :v2RecordProcessorFactory, :recordProcessorFactory, [Java::JavaClass.for_name(KCL_PROCESSOR_FACTORY_CLASS)]
>
> Haha, I know the feeling. JRuby isn't my forte either. I think I see what
> you are saying now, though it seems like if we updated some of the tests,
> this could be avoided... I'll take a stab at that later.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/codekitchen/logstash-input-kinesis/pull/4/files#r40595584>
> .
>